Libraries/LibJS/Runtime/StringPrototype.cpp

```diff
@@ -307,4 +308,22 @@ Value StringPrototype::trim_end(Interpreter& interpreter)
     return trim_string(interpreter, *this_object, TrimMode::Right);
 }
 
+Value StringPrototype::concat(Interpreter& interpreter)
+{
+    auto* this_object = interpreter.this_value().to_object(interpreter.heap());
+    if (!this_object)
+        return {};
+    auto& string = this_object->to_string().as_string()->string();
+
+    StringBuilder builder;
+    builder.append(string);
+
+    for(size_t i = 0; i < interpreter.argument_count(); ++i) {
```

Comment:
Please use `clang-format`. Also please squash commits instead of stacking them, per the contribution guidelines.